src/Sparrow/Collections/LockFreeRingBuffer.cs

```diff
@@ -0,0 +1,160 @@
+﻿using System;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Threading;
+using Sparrow.Binary;
+
+namespace Sparrow.Collections
+{
+    public sealed class LockFreeRingBuffer<T>
+    {
+        // Explicit layout to control the exact memory layout
+        [StructLayout(LayoutKind.Sequential, Size = 128)]
+        internal unsafe struct Cell
+        {
+            // Sequence number at offset 0
+            internal long Sequence;
+
+            private fixed byte _sequencePadding[64 - sizeof(long)];
+
+            // Value at offset 64 to ensure it's on a different cache line
+            internal T Value;
```

Comment:
what if `T` here is a large struct? would that might be a problem, give that the size of `Cell` is 128?